### PR TITLE
[TFLite] Add int8 and int16x8 support for LOG operator

### PR DESCRIPTION
**Pull request by Tessil**
Hello,

This PR adds int8 and int16 quantization support for the LOG operator using a lookup table.

Thanks,
Thibaut

### CODE REVIEW

File: tensorflow/lite/testing/op_tests/elementwise.py [hunk 1]
```diff
@@ -77,7 +78,8 @@ def make_sin_tests(options):
 @register_make_test_function()
 def make_log_tests(options):
   """Make a set of tests to do log."""
-  return _make_elementwise_tests(tf.math.log)(options)
+  return _make_elementwise_tests(tf.math.log, allow_fully_quantize=True,
+                                 min_value=.1, max_value=1)(options)
```
Comment (grantjensen) [inline tensorflow/lite/testing/op_tests/elementwise.py:82]: Why do you set the max value to 1 here?

**Review by grantjensen — APPROVED** [review]

File: tensorflow/lite/testing/op_tests/elementwise.py (hunk 1 above)
Comment (Tessil) [reply tensorflow/lite/testing/op_tests/elementwise.py:82]: It's a bit arbitrary as I used the same range as the rsqrt test. I could leave it to the default 100 value if it's preferred.

Comment (grantjensen) [reply tensorflow/lite/testing/op_tests/elementwise.py:82]: Yeah, please leave the default value. Everything else looks good though!

Comment (Tessil) [reply tensorflow/lite/testing/op_tests/elementwise.py]: Thanks, I updated it to 100.

**Review by grantjensen — APPROVED** [review]

**Comment by grantjensen** [comment]
Hi @Tessil, one error left to fix. The E2E tests (whose params are in part generated by elementwise.py) are failing. Specifically when quant_16x8=True for rsqrt fails with "Type INT16 is unsupported by op Rsqrt." Then, some of the log tests are generating different numbers from what is expected. Let me know if you have any troubles reproducing, thanks.

**Comment by Tessil** [comment]
Hi @grantjensen I wasn't able to reproduce the issues.

I ran the following two commands with success:
`bazel run //tensorflow/lite/testing:generate_examples -- --zip_to_output=log -- --output_path=test`
`bazel test //tensorflow/lite/kernels:elementwise_test`

I have updated the PR though to remove QI16 from the MLIR TFL_LogOp definition as it's not necessary currently (no official support for int16x8 MLIR quantizer).

**Review by grantjensen — APPROVED** [review]

**Comment by grantjensen** [comment]
` bazel test //third_party/tensorflow/lite/testing:zip_test_log`
 ` bazel test //third_party/tensorflow/lite/testing:zip_test_rsqrt`

**Pull request merged**